src/neptune/internal/utils/uncaught_exception_handler.py

```diff
@@ -21,59 +21,59 @@
 import uuid
 from platform import node as get_hostname
 from typing import (
-    TYPE_CHECKING,
+    Any,
     Callable,
     List,
 )
 
 from neptune.internal.utils.logger import get_logger
 
-if TYPE_CHECKING:
-    pass
-
 _logger = get_logger()
 
 
 class UncaughtExceptionHandler:
-    def __init__(self):
+    def __init__(self) -> None:
         self._previous_uncaught_exception_handler = None
         self._handlers = dict()
         self._lock = threading.Lock()
 
-    def activate(self):
-        with self._lock:
-            this = self
-
-            def exception_handler(exc_type, exc_val, exc_tb):
-                header_lines = [
-                    f"An uncaught exception occurred while run was active on worker {get_hostname()}.",
-                    "Marking run as failed",
-                    "Traceback:",
-                ]
-
-                traceback_lines = header_lines + traceback.format_tb(exc_tb) + str(exc_val).split("\n")
-                for _, handler in self._handlers.items():
-                    handler(traceback_lines)
+    def trigger(self, exc_type, exc_val, exc_tb) -> None:
+        header_lines = [
+            f"An uncaught exception occurred while run was active on worker {get_hostname()}.",
+            "Marking run as failed",
+            "Traceback:",
+        ]
 
-                this._previous_uncaught_exception_handler(exc_type, exc_val, exc_tb)
+        traceback_lines = header_lines + traceback.format_tb(exc_tb) + str(exc_val).split("\n")
+        for _, handler in self._handlers.items():
+            handler(traceback_lines)
 
-            if self._previous_uncaught_exception_handler is None:
-                self._previous_uncaught_exception_handler = sys.excepthook
-                sys.excepthook = exception_handler
+    def activate(self) -> None:
+        with self._lock:
+            if self._previous_uncaught_exception_handler is not None:
+                return
```

Comment:
I had to verify it but looks like `return` could be called safely from context-manager: https://stackoverflow.com/a/9885287